src/python/espressomd/analyze.pyx

```diff
@@ -139,13 +139,16 @@ class Analysis(object):
         if id != None:
             if not isinstance(id, int):
                 raise ValueError("Id has to be an integer")
+            if id >= len(self._system.part):
```

Comment:
That's not a good test... you can set a single particle with whatever id you want it to have

Reply:
Better use sth like: if not id in system.part[:].id

Reply:
good suggestion, I somehow didn't make the realization that the ID property and index were different.